engine.py

```diff
@@ -625,7 +598,7 @@ def show_modal(self, title, bundle, widget_class, *args, **kwargs):
                 self._parent_to_max = True
 
             # finally launch it, modal state
-            status = dialog.exec_()
+            status = dialog.exec()
```

Comment:
This rings a bell as it's important in PySide2. Why do we need this change?

Reply:
It's PySide 1 syntax. Since we're starting to deprecate Qt4 we can use just `exec`. 

Note: `exec_` throws warnings in PySide6.

Reply:
I remember now where I saw that:
- https://github.com/shotgunsoftware/tk-core/blob/v0.21.2/python/tank/authentication/login_dialog.py#L356
- https://github.com/shotgunsoftware/tk-core/blob/v0.21.2/python/tank/authentication/login_dialog.py#L662